src/govuk/components/details/details.mjs

```diff
@@ -15,25 +15,40 @@ var KEY_SPACE = 32
  * Details component
  *
  * @class
- * @param {HTMLElement} $module - HTML element to use for details
+ * @param {HTMLDetailsElement | HTMLElement | Element} $module - HTML element to use for details
+ * @this {Details}
  */
 function Details ($module) {
+  if (!($module instanceof HTMLElement)) {
+    // Return instance for method chaining
+    // using `new Details($module).init()`
+    return this
+  }
+
   this.$module = $module
 }
 
+/**
+ * Initialise component
+ *
+ * @returns {Details} Details component
+ */
 Details.prototype.init = function () {
+  // Check that required elements are present
   if (!this.$module) {
-    return
+    return this
   }
 
   // If there is native details support, we want to avoid running code to polyfill native behaviour.
-  var hasNativeDetails = typeof this.$module.open === 'boolean'
+  var hasNativeDetails = this.$module instanceof HTMLDetailsElement
```

Comment:
**issue**: We'll need to check that this still works in older browsers (maybe accessing via `window.HTMLDetailsElement` to avoid complaining that `HTMLDetailsElement` is not defined).

Reply:
Moved to:

* https://github.com/alphagov/govuk-frontend/pull/3123